File: bubble-sort/bubble.py
#!/usr/bin/env python

def bubble(arr):
    n = len(arr)
    for i in range(n):
        for j in range(n-1):
            if arr[j] > arr[j+1]:
                arr[j] = arr[j] + arr[j+1];
                arr[j+1] = arr[j] - arr[j+1];
                arr[j] = arr[j] - arr[j+1];


n = int(raw_input(""))

arr = range(n)

for i in range(n):
    arr[i] = int(raw_input(""))

bubble(arr)

for i in arr:
    print i,

File: bubble-sort/bubble.rb
#!/usr/bin/ruby

def bubble(arr)
    for i in 0...arr.length
        for j in 0...arr.length-1
            if (arr[j] > arr[j+1])
                arr[j] = arr[j] + arr[j+1];
                arr[j+1] = arr[j] - arr[j+1];
                arr[j] = arr[j] - arr[j+1];
            end
        end
    end
end

n = Integer(gets)

arr = Array.new(n, 0)

for i in 0...n
    arr[i] = Integer(gets)
end

bubble(arr)

for i in arr
    puts i
end

